eos_data_distribution/ndn/dbus/base.py

```diff
@@ -184,9 +184,9 @@ def _on_manager_ready(self, proxy, res):
 
     def flush_pending_interests(self):
         logger.debug('processing pending interests: %s', self._pending_interests.keys())
-        for interest, dbus_path, dbus_name in self._pending_interests.values():
-             if self._dbus_express_interest(interest, dbus_path, dbus_name):
-                del self._pending_interests[interest]
+        for interest in self._pending_interests.keys():
```

Comment:
You could get the keys beforehand, since you use it twice in this function (or move the debug inside the loop)